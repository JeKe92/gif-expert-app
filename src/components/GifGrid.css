.gifs-grid {
    display: grid;
    gap: 20px;
    grid-template-columns: repeat(4, 1fr);

}
@media screen and ( max-width: 960px ) {
    .gifs-grid {
        grid-template-columns: repeat(3, 1fr);
    }
}
@media screen and ( max-width: 768px ) {
    .gifs-grid {
        grid-template-columns: repeat(2, 1fr);
    }
}
@media screen and ( max-width: 550px ) {
    .gifs-grid {
        grid-template-columns: repeat(1, 1fr);
    }
}