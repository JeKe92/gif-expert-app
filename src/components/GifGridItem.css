.gifs-card{
    align-self: center;
    border: 2px solid #eeeeee;
    border-radius: 5px;
    height: auto;
    justify-self: center;
    min-height: 100%;
    width: 100%;    
}
.gifs-card>img {
    display: block;
    height: 200px;
    margin-left:auto;
    margin-right: auto;
    max-width: 100%;
}
.gifs-card>figcaption {
    text-align:center;
    padding: 10px;
    text-transform: capitalize;
}
