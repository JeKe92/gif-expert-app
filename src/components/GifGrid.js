import React from 'react';
import PropTypes from 'prop-types';
import { useFetchHook } from '../hooks/useFetchHook';
import './GifGrid.css';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({category}) => {

    const {data: images, loading} = useFetchHook(category);    

    return (
        <article>
            <h3 className='animate__animated animate__fadeInLeft'>{category} {loading && <small>"Cargando..."</small>}</h3>
            <div className="gifs-grid">
            {images.map( gif => <GifGridItem key={gif.id} {...gif}/>)}
            </div>
        </article>
    );
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired,
};