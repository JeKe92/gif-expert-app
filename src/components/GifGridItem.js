import React from 'react';
import PropTypes from 'prop-types';

import './GifGridItem.css';

export const GifGridItem = ( { title, url } ) => (
            <figure className="gifs-card animate__animated animate__fadeInDown">
                            <img src={url} alt={title}></img>
                            <figcaption>{title}</figcaption>
            </figure>
);

GifGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
}