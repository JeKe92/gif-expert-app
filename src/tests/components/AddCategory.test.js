import React from 'react';
import { shallow } from "enzyme";
import { AddCategory } from "../../components/AddCategory";

describe('Tests for Add Categories component', () => {
    const setCategories = jest.fn();
    let wrapper = shallow(<AddCategory setCategories={setCategories}/>);

    beforeEach(()=>{
        jest.clearAllMocks();
        wrapper = shallow(<AddCategory setCategories={setCategories}/>);
    });

    test('should show the component correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('should change by using the text input', () => {
        const input = wrapper.find('input');
        const value = 'Hola mundo';
        input.simulate('change', {target: {value}});
    });

    test('should not call the setCategories function', () => {
        wrapper.find('form').simulate('submit', {preventDefault(){}});
        expect( setCategories ).not.toHaveBeenCalled();
    });
    
    test('should call the function and clear the inputValue', () => {
        wrapper.find('input').simulate('change', {target:{value: 'Hola Mundo'}});
        wrapper.find('form').simulate('submit', {preventDefault(){}});
        expect( setCategories ).toHaveBeenCalled();
        expect( setCategories ).toHaveBeenCalledWith( expect.any(Function) );
        expect( wrapper.find('input').prop('value')).toBe('');
    });
    
});
