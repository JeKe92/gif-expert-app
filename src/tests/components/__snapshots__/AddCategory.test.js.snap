// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for Add Categories component should show the component correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <label
    htmlFor="buscar"
  >
    <input
      id="buscar"
      onChange={[Function]}
      type="text"
      value=""
    />
  </label>
</form>
`;
